doc/guides/west/manifest.rst

```diff
@@ -1179,6 +1189,450 @@ processed in this order:
+isn't relevant except to make the examples complete and self-contained.
+
+Example 1: no disabled groups
+-----------------------------
+
+The entire manifest file is:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: foo
+         groups:
+           - groupA
+       - name: bar
+         groups:
+           - groupA
+           - groupB
+       - name: baz
+
+     defaults:
+       remote: example-remote
+     remotes:
+       - name: example-remote
+         url-base: https://git.example.com
+
+The ``manifest.groups`` configuration option is not set (you can ensure
+this by running ``west config -D manifest.groups``).
+
+No groups are disabled, because all groups are enabled by default. Therefore,
+all three projects (``foo``, ``bar``, and ``baz``) are active. Note that there
+is no way to make project ``baz`` inactive, since it has no groups.
+
+Example 2: Disabling one group via manifest
+-------------------------------------------
+
+The entire manifest file is:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: foo
+         groups:
+           - groupA
+       - name: bar
+         groups:
+           - groupA
+           - groupB
+
+     groups: [-groupA]
+
+     defaults:
+       remote: example-remote
+     remotes:
+       - name: example-remote
+         url-base: https://git.example.com
+
+The ``manifest.groups`` configuration option is not set (you can ensure
+this by running ``west config -D manifest.groups``).
+
+Since ``groupA`` is disabled, project ``foo`` is inactive. Project ``bar`` is
+active, because ``groupB`` is enabled.
+
+Example 3: Disabling multiple groups via manifest
+-------------------------------------------------
+
+The entire manifest file is:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: foo
+         groups:
+           - groupA
+       - name: bar
+         groups:
+           - groupA
+           - groupB
+
+     groups: [-groupA,-groupB]
+
+     defaults:
+       remote: example-remote
+     remotes:
+       - name: example-remote
+         url-base: https://git.example.com
+
+The ``manifest.groups`` configuration option is not set (you can ensure
+this by running ``west config -D manifest.groups``).
+
+Both ``foo`` and ``bar`` are inactive, because all of their groups are
+disabled.
+
+Example 4: Disabling a group via configuration
+----------------------------------------------
+
+The entire manifest file is:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: foo
+         groups:
+           - groupA
+       - name: bar
+         groups:
+           - groupA
+           - groupB
+
+     defaults:
+       remote: example-remote
+     remotes:
+       - name: example-remote
+         url-base: https://git.example.com
+
+The ``manifest.groups`` configuration option is set to ``-groupA`` (you can
+ensure this by running ``west config manifest.groups -- -groupA``; the extra
```

Comment:
It seems to me that `--` parameter is mandatory when disabling a group but obligatory when enabling it.
Can we unify group enable and disable commands in terms of adding `--` ?

Reply:
> It seems to me that `--` parameter is mandatory when disabling a group but obligatory when enabling it.

I don't understand why you say it is "obligatory" when enabling. Did you mean to say "optional"?

> Can we unify group enable and disable commands in terms of adding `--` ?

I'm not sure what you're asking for. The extra `--` is just a POSIX thing to make sure that the string "-groupA" in the command line is treated as a positional argument.

https://unix.stackexchange.com/questions/11376/what-does-double-dash-mean

You could write `west config manifest.groups -- +foo` and it would set `manifest.groups` to `+foo`, but it's not necessary to use `--` in this case because `+foo` is not ambiguous: it is definitely a positional argument because it does not begin with `-`.

Reply:
What I was asking about using the `west config manifest.groups -- +foo` command for adding group and  `west config manifest.groups -- -foo` for removing. It just seems more coherent to me. But the actual description is also good.

Reply:
I see, thanks for explaining.

I try to avoid using `--` in `west config` lines in the documentation unless they are really necessary, because I don't want people to think that the ugly syntax is always required. The `--` is only needed when dealing with strings that start with `-`, so that is where I'd like to keep it. This is consistent with other places in the documentation, like here:

https://docs.zephyrproject.org/latest/guides/west/build-flash-debug.html#permanent-cmake-arguments